libs/report/committers/src/lib/parse-monthly.ts

```diff
@@ -1,44 +1,29 @@
-import {
-  eachMonthOfInterval,
-  format,
-  isWithinInterval,
-  max,
-  min,
-} from 'date-fns';
-import { MonthlyData } from './types';
-
-export function parseMonthly(
-  startDate: Date,
-  endDate: Date,
-  entries: { commitHash: string; committer: string; date: string }[]
-) {
+import { eachMonthOfInterval, format, isWithinInterval } from 'date-fns';
+import { Commit, MonthlyData } from './types';
+
+export function parseMonthly(startDate: Date, endDate: Date, entries: Commit[]) {
   const monthly: MonthlyData[] = [];
-  const dates = [startDate, endDate];
-  const ival = {
-    start: min(dates),
-    end: max(dates),
-  };
-  const range = eachMonthOfInterval(ival);
-  for (const idxr in range) {
-    const idx = parseInt(idxr);
-    if (idx + 1 >= range.length) {
-      continue;
-    }
-    const [start, end] = [range[idx], range[idx + 1]];
+  const range = eachMonthOfInterval({
+    start: startDate,
+    end: endDate,
+  });
+
+  for (let idx = 0; idx < range.length - 1; idx++) {
+    // `convertToUTC` is an awful hack to convert the date to UTC
+    // This is needed because the date-fns eachMonthOfInterval function
+    // is not timezone aware and will return the date in the local timezone
+    // Refs: https://github.com/date-fns/date-fns/issues/3097
+    const [start, end] = [convertToUTC(range[idx]), convertToUTC(range[idx + 1])];
     const month: MonthlyData = {
-      name: format(start, 'LLLL yyyy'),
+      month: format(start, 'LLLL yyyy'),
       start,
       end,
       committers: {},
     };
 
     for (const rec of entries) {
-      if (isWithinInterval(new Date(rec.date), { start, end })) {
-        month.committers[rec.committer] = month.committers[rec.committer] || [];
-        month.committers[rec.committer].push({
-          hash: rec.commitHash,
-          date: rec.date,
-        });
+      if (isWithinInterval(rec.date, { start, end })) {
```

Comment:
I don't think this is correct. The entry's `rec.date` has been retrieved by git based on the unaltered start-/end-dates. Here, we are comparing them with the modifed `start`/`end` dates. Some commits could get misaccounted for.

Reply:
Not sure what you mean? We are checking if commit date X is between date Y and Z